apps/src/code-studio/components/Libraries/LibraryClientApi.js

```diff
@@ -61,8 +63,53 @@ export default class LibraryClientApi {
     });
   }
 
-  getClassLibraries(onSuccess, onError) {
-    // TODO: find all libraries published by other students in a classroom
-    return undefined;
+  async getClassLibraries(onSuccess, onError) {
+    getUserSections(sections => {
+      // TODO: Add backend controller action so this doesn't require multiple AJAX requests or this super clunky promise structure.
+      let requests = [];
+      let allLibraries = [];
+      let promises = [];
+      let libraryIds = {};
+      sections.forEach(section => {
+        requests.push(
+          $.ajax({
```

Comment:
Tiny style nit: should we be using `fetch` here? (we polyfill it via babel)

Reply:
Good point. I'm going to leave this for now and fix it when I do the second pass with the new controller action.